src/go/rpk/pkg/config/weak.go

```diff
@@ -0,0 +1,595 @@
+	}
+	if err := n.Decode(&internal); err != nil {
+		return err
+	}
+	rpc.Directory = string(internal.Directory)
+	rpc.ID = int(internal.ID)
+	rpc.Rack = string(internal.Rack)
+	rpc.SeedServers = internal.SeedServers
+	rpc.RPCServer = internal.RPCServer
+	rpc.RPCServerTLS = internal.RPCServerTLS
+	rpc.KafkaAPI = internal.KafkaAPI
+	rpc.KafkaAPITLS = internal.KafkaAPITLS
+	rpc.AdminAPI = internal.AdminAPI
+	rpc.AdminAPITLS = internal.AdminAPITLS
+	rpc.CoprocSupervisorServer = internal.CoprocSupervisorServer
+	rpc.AdminAPIDocDir = string(internal.AdminAPIDocDir)
+	rpc.DashboardDir = string(internal.DashboardDir)
+	rpc.CloudStorageCacheDirectory = string(internal.CloudStorageCacheDirectory)
+	rpc.AdvertisedRPCAPI = internal.AdvertisedRPCAPI
+	rpc.AdvertisedKafkaAPI = internal.AdvertisedKafkaAPI
+	rpc.DeveloperMode = bool(internal.DeveloperMode)
+	rpc.Other = internal.Other
+	return nil
+}
+
+func (rpkc *RpkConfig) UnmarshalYAML(n *yaml.Node) error {
+	var internal struct {
+		// Deprecated 2021-07-1
+		TLS *TLS `yaml:"tls"`
+		// Deprecated 2021-07-1
+		SASL *SASL `yaml:"sasl"`
+
+		KafkaAPI                 RpkKafkaAPI     `yaml:"kafka_api"`
+		AdminAPI                 RpkAdminAPI     `yaml:"admin_api"`
+		AdditionalStartFlags     weakStringArray `yaml:"additional_start_flags"`
+		EnableUsageStats         weakBool        `yaml:"enable_usage_stats"`
+		TuneNetwork              weakBool        `yaml:"tune_network"`
+		TuneDiskScheduler        weakBool        `yaml:"tune_disk_scheduler"`
+		TuneNomerges             weakBool        `yaml:"tune_disk_nomerges"`
+		TuneDiskWriteCache       weakBool        `yaml:"tune_disk_write_cache"`
+		TuneDiskIrq              weakBool        `yaml:"tune_disk_irq"`
+		TuneFstrim               weakBool        `yaml:"tune_fstrim"`
+		TuneCPU                  weakBool        `yaml:"tune_cpu"`
+		TuneAioEvents            weakBool        `yaml:"tune_aio_events"`
+		TuneClocksource          weakBool        `yaml:"tune_clocksource"`
+		TuneSwappiness           weakBool        `yaml:"tune_swappiness"`
+		TuneTransparentHugePages weakBool        `yaml:"tune_transparent_hugepages"`
+		EnableMemoryLocking      weakBool        `yaml:"enable_memory_locking"`
+		TuneCoredump             weakBool        `yaml:"tune_coredump"`
+		CoredumpDir              weakString      `yaml:"coredump_dir"`
+		TuneBallastFile          weakBool        `yaml:"tune_ballast_file"`
+		BallastFilePath          weakString      `yaml:"ballast_file_path"`
+		BallastFileSize          weakString      `yaml:"ballast_file_size"`
+		WellKnownIo              weakString      `yaml:"well_known_io"`
+		Overprovisioned          weakBool        `yaml:"overprovisioned"`
+		SMP                      *weakInt        `yaml:"smp"`
+	}
+	if err := n.Decode(&internal); err != nil {
+		return err
+	}
+	if internal.TLS != nil {
+		fmt.Println("redpanda.yaml: specifying Kafka TLS fields in rpk section is deprecated and will be removed in the future")
```

Comment:
How about `redpanda.yaml: rpk.tls has been replaced with rpk.kafka_api.tls and will be removed in the future`

and for below, `rpk.sasl`?